circle.yml

```diff
@@ -3,23 +3,26 @@ machine:
     - docker
   environment:
     CLOUDSDK_CORE_DISABLE_PROMPTS: 1
-    image_name: codeclimate-rubocop
-
-dependencies:
-  pre:
-    - echo $gcloud_json_key_base64 | sed 's/ //g' | base64 -d > /tmp/gcloud_key.json
-    - curl https://sdk.cloud.google.com | bash
-    - gcloud auth activate-service-account $gcloud_account_email --key-file /tmp/gcloud_key.json
-    - gcloud docker -a
+    PRIVATE_REGISTRY: us.gcr.io/code_climate
+    PUBLIC_REGISTRY: hub.docker.com/codeclimate
 
 test:
   override:
     - bundle exec rake
-    - docker build -t=$registry_root/$image_name:b$CIRCLE_BUILD_NUM .
+    - docker build -t=$PRIVATE_REGISTRY/$CIRCLE_PROJECT_REPONAME:b$CIRCLE_BUILD_NUM .
 
 deployment:
   registry:
     branch: master
+    owner: codeclimate
     commands:
-      - docker push $registry_root/$image_name:b$CIRCLE_BUILD_NUM
-
+     - echo $GCLOUD_JSON_KEY_BASE64 | sed 's/ //g' | base64 -d > /tmp/gcloud_key.json
+      - curl https://sdk.cloud.google.com | bash
+      - gcloud auth activate-service-account $GCLOUD_ACCOUNT_EMAIL --key-file /tmp/gcloud_key.json
+      - gcloud docker -a
+      - git tag b$CIRCLE_BUILD_NUM master && git push origin b$CIRCLE_BUILD_NUM
+      - docker push $PRIVATE_REGISTRY/$CIRCLE_PROJECT_REPONAME:b$CIRCLE_BUILD_NUM
+      - docker tag $PRIVATE_REGISTRY/$CIRCLE_PROJECT_REPONAME:b$CIRCLE_BUILD_NUM $PUBLIC_REGISTRY/$CIRCLE_PROJECT_REPONAME:b$CIRCLE_BUILD_NUM
+      - docker push $PUBLIC_REGISTRY/$CIRCLE_PROJECT_REPONAME:b$CIRCLE_BUILD_NUM
+      - docker tag -f $PUBLIC_REGISTRY/$CIRCLE_PROJECT_REPONAME:b$CIRCLE_BUILD_NUM $PUBLIC_REGISTRY/$CIRCLE_PROJECT_REPONAME:latest
```

Comment:
I realized I was pushing the tag but not `latest` for the public docker hub registry. Since that's what the CLI actually uses, I think we need to push both?